File: zhaoxin_ws_fangzhen/src/practice/scripts/property_listener.py
#!/usr/bin/env python3

import rospy
from practice.msg import my_property

def callback(data):
    rospy.loginfo("Received message: %s", data.name)
    rospy.loginfo("Received message: %s", data.Class)

###补全回调函数###

def property_listener():
    rospy.init_node('property_listener')
    rospy.Subscriber("property", my_property, callback)
    rospy.spin()

if __name__ == '__main__':
    property_listener()

File: zhaoxin_ws_fangzhen/src/practice/src/property_listener.cpp
#include <ros/ros.h>
#include <practice/my_property.h>

void callback(const practice::my_property::ConstPtr& msg) {
    ROS_INFO("Received message: %s", msg->name.c_str());
    ROS_INFO("Received message: %d", msg->Class);
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "property_listener");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("property", 10, callback);
    ros::spin();
    return 0;
}